### ISSUE TITLE
Question: How to time out a connection?

### ISSUE THREAD

**Issue by Icedude907** [uniciant/just-webrtc#3]
Hi there. Thanks for making this library - its been really useful.

If peer A/the client closes their browser unexpectedly - I understand that the `iceConnectionState`  variable will become `disconnected` on peer B/the server. Is there a way within the library to detect that and thus destroy the connection if, say, the client disconnects for more than 15 seconds?

Thanks again

**Comment by poshcoe** [comment]
> Hi there. Thanks for making this library - its been really useful.

Hi, glad to hear it!

> If peer A/the client closes their browser unexpectedly - I understand that the `iceConnectionState` variable will become `disconnected` on peer B/the server. Is there a way within the library to detect that and thus destroy the connection if, say, the client disconnects for more than 15 seconds?

No there isn't and this is an oversight on my part! `PeerConnectionState` is not exposed properly in the current API. My current thinking is to expose a new `PeerConnectionExt` method:

```rust
async fn state_change(&self) -> PeerConnectionState;
```

This method could then be monitored concurrently to operating connected WebRTC channels, and would enable the user to detect state changes and act accordingly. The `signalling-peer` example should also be updated such that it implements peer monitoring and timeout.

I'll look into this and open a PR soon.

**Comment by Icedude907** [comment]
Sounds good, thanks!

**Comment by poshcoe** [comment]
Hi @Icedude907,

I have found some time for this today and just made a release which incorporates the changes as mentioned, hope this helps!

The `signalling-peer` example now implements a single threaded connection monitoring and recovery task, with timeout. With my limited testing I have found it to be significantly more robust without resorting to the cost of opening a new connection.

A few methods also dropped the mutability requirement, which may result in simplification of code on the user side.

Let me know if you have any other questions or suggestions :grin:
